src/lib/interp.rs

```diff
@@ -238,22 +257,54 @@ impl VM {
         self.pc = self.bytecode.labels.get(
             &(GLOBAL_NSPACE.to_string(), MAIN_FN.to_string()))
             .expect("Main method not found").clone();
-        self.frames.push(Frame::new(Vec::new(), self.bytecode.bytecode.len()))
+        self.frames.push(Frame::new("main".to_string(), Vec::new(), self.bytecode.bytecode.len()))
+    }
+
+    fn unroll_if_raised(&mut self) {
+        if self.frames.last().unwrap().raise {
+            let mut backtrace: Vec<NativeType> = Vec::new();
+            let mut try_index: usize = self.frames.len() - 1;
+            for (i, f) in self.frames.iter().rev().enumerate() {
+                if f.in_try {
+                    try_index = i;
+                    break
+                }
+                else {
+                    backtrace.push(NativeType::Str(f.name.to_string()));
+                }
+            }
+            let try_index = (self.frames.len() -1) - try_index;
+            self.frames.drain(try_index..);
```

Comment:
You probably want `self.frames.clear()` here. [You might also want to think about `shrink_to_fit`, but it's probably not appropriate here.]

Reply:
I'm not sure about that. Clear removes all the values, I only want to remove the values up-to the next appropriate try/catch block. This is why I iterate over the frame stack before to work out where to break.

Reply:
You're quite right.